### PR TITLE
Proposed edit to CI/CD

### PR DESCRIPTION
**Pull request by fkrjakDatasentics**
Currently supported only DevOps -> ON PR creation Check Config on all Commits to Main (Should be always protected for direct commits) start rechecking of export configs.

### CODE REVIEW

File: _cicd/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        steps:
+        - task: CmdLine@2
+          inputs:
+            script: 'pip3 install databricks-cli==0.17.3'
+        - task: PythonScript@0
+          inputs:
+            scriptSource: 'filePath'
+            scriptPath: '$(System.DefaultWorkingDirectory)/_cicd/check_jobs.py'
+            arguments: '$(TOKEN) $(HOST) $(TAG) "$(CLUSTER)"'
+            workingDirectory: '$(System.DefaultWorkingDirectory)'
```
Comment (guderkar) [inline _cicd/azure-pipelines.yml:43]: why tag and cluster are defined as devops variables? Why shouldn't we define the whole databricks workflow in yaml/json?

File: _cicd/check_config.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import yaml
+
+
+def check_destinations_exist(config):
+    for i in config["exports"].keys():
+        if not config["exports"][i]["destination"] in config["destinations"].keys():
+            raise Exception
```
Comment (guderkar) [inline _cicd/check_config.py:7]: add exception message

**Review by guderkar — CHANGES_REQUESTED** [review]

File: _cicd/check_config.py (hunk 2 above)
Comment (fkrjakDatasentics) [reply _cicd/check_config.py]: added in new commit

File: _cicd/azure-pipelines.yml (hunk 1 above)
Comment (fkrjakDatasentics) [reply _cicd/azure-pipelines.yml:42]: fixed in commit

File: config.yaml [hunk 3]
```diff
@@ -43,3 +58,26 @@ parameters:
...
+  cluster:
+    spark_version: "10.4.x-scala2.12"
+    spark_conf:
+      spark.databricks.delta.preview.enabled: "true"
+      fs.azure.account.oauth2.client.id.odapczlakeg2test.dfs.core.windows.net: "{{secrets/unit-kv/dbx-client-id}}"
+      spark.client_secret: "{{secrets/unit-kv/dbx-client-secret}}"
+      fs.azure.account.oauth.provider.type.odapczlakeg2test.dfs.core.windows.net: "org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider"
+      fs.azure.account.auth.type.odapczlakeg2test.dfs.core.windows.net: "OAuth"
+      fs.azure.account.oauth2.client.secret.odapczlakeg2test.dfs.core.windows.net: "{{secrets/unit-kv/dbx-client-secret}}"
+      fs.azure.account.oauth2.client.endpoint.odapczlakeg2test.dfs.core.windows.net: "https://login.microsoftonline.com/585127aa-8691-4f25-9777-8cb78ccf0d91/oauth2/token"
```
Comment (guderkar) [inline config.yaml:73]: We don't need any external storage for demo so this conf is not needed

File: config.yaml [hunk 4]
```diff
@@ -43,3 +58,26 @@ parameters:
...
+    custom_tags:
+      PxPorpuse: "ExportImport"
+    spark_env_vars:
+      READ_ENV: test
+      APP_ENV: test
+      TENANT_ID: "585127aa-8691-4f25-9777-8cb78ccf0d91"
```
Comment (guderkar) [inline config.yaml:79]: Those env vars are also not needed

File: config.yaml [hunk 5]
```diff
@@ -43,3 +58,26 @@ parameters:
...
+    instance_pool_id: "0712-081441-voile3-pool-dt4ut8z6"
```
Comment (guderkar) [inline config.yaml:80]: Don't assume pool will be present, just define classical job cluster instead

**Review by guderkar — CHANGES_REQUESTED** [review]

**Review by guderkar — APPROVED** [review]

**Pull request merged**